### PR TITLE
feat: Registering transformers after init()

### PR DESCRIPTION
**Pull request by Luukvdm**
We are trying to extend Move2Kube and want to create some specific transformers in Golang. But the code currently doesn't allow that. This PR adds a `RegisterTransformer` method that can used to add transformers from outside the module and helps us achieve our goal.
We also did consider the Starlark approach, but found this to limited.

### CODE REVIEW

**Comment by ashokponkumar** [comment]
Thanks @Luukvdm for the PR. Looking forward to your contributions to Move2Kube. 

Can you please help us understand the problems this PR solve and how it solves it. 

I understand that you want to write custom transformers in golang, as you have rightly mentioned Move2Kube currently has the Starlark based approach and external Transformer (command line executable/containers) only. But I don't under how the newly added Register helps, since it does not seem to be called at this point. Is it being used as a library?

If you can help us understand the overall use case, we can together architect it, shape this PR and get it merged.

Looking forward to shaping the PR together.

**Comment by Luukvdm** [comment]
Yes, we would rather not maintain a fork, so we would like to have a main that imports Move2Kube modules and start the rootcmd like normal. But also register some custom transformers like this:

```go
func main() {
	rootCmd := cmd.GetRootCmd()
        // Load Move2Kube assets and our own custom assets
	tempPath, err := loadAssets()
	if err != nil {
		logrus.Fatalf("failed to load assets: %s\n", err)
	}
	// Load our transformers
	tf := new(transformers.OurCustomTransformer)
	err = transformer.AddTransformer(tf)
	if err != nil {
		logrus.Errorf("ruh roh")
	}

	defer os.RemoveAll(tempPath)
	if err := rootCmd.Execute(); err != nil {
		logrus.Fatalf("Error: %q", err)
	}
}
```
So aside from the `main.go` file, we only need assets and custom transformers in our project.

**Review by ashokponkumar — APPROVED** [review]
Got it @Luukvdm ! Looks good. Please feel free to feed in your inputs on any more improvements we can make to Move2Kube and we welcome new PRs and features.

**Comment by ashokponkumar** [comment]
@Luukvdm 
Can you please do the following:
1. Can you change the title to `feat: ...`
2. Also please fix the DCO (needs `sign-off`) - https://github.com/konveyor/move2kube/pull/807/checks?check_run_id=6414233812 has steps on how to fix
3. Rebase the main branch with the main of konveyor

Once we are done with this, we can merge the PR.

**Comment by ashokponkumar** [comment]
@Luukvdm [Konveyor](https://www.konveyor.io/) is now a [CNCF Sandbox project](https://www.konveyor.io/blog/konveyor-is-a-cncf-sandbox-project/). We are doing an an exercise to understand how the tool is being used, to help shape the future direction and priorities of the tool. 

If you are comfortable sharing, can you please help us understand the usecase Move2Kube solves for you, and whether it helped in your usecase. Also, if you can spare a few mins to answer the below survey it will help.

https://airtable.com/shrD7qbYrSMe33Gia

**Pull request merged**
